src/frequenz/client/electricity_trading/_client.py

```diff
@@ -49,6 +50,32 @@ class _Sentinel:
 NO_VALUE = _Sentinel()
 
 
+def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> None:
+    """
+    Validate that the decimal places of a given value do not exceed a specified limit.
+
+    Args:
+        value: The value to be checked.
+        decimal_places: The maximum allowed decimal places.
+        name: The name of the value (for error messages).
+
+    Raises:
+        ValueError: If the value has more decimal places than allowed.
+        ValueError: If the value is not a valid decimal number.
+    """
+    assert decimal_places >= 0, "The decimal places must be a non-negative integer."
+
+    if value is not None:
```

Comment:
The type-checker should alert us calling ` validate_decimal_places()` with `None` as a value as the function only accepts a `Decimal` type. So if this function can be called with a None, then the signature should be `def validate_decimal_places(value: Decimal | None, decimal_places...`, otherwise this check is not needed.

Reply:
> The type-checker should alert us calling  validate_decimal_places() with None

oh yea right I forgot that I'm already checking it before calling the function my bad